agent/src/main/java/com/criteo/hadoop/garmadon/agent/modules/PrestoModule.java

```diff
@@ -0,0 +1,41 @@
+package com.criteo.hadoop.garmadon.agent.modules;
+
+import com.criteo.hadoop.garmadon.agent.AsyncEventProcessor;
+import com.criteo.hadoop.garmadon.agent.headers.StandaloneHeader;
+import com.criteo.hadoop.garmadon.agent.tracers.hadoop.hdfs.FileSystemTracer;
+import com.criteo.hadoop.garmadon.agent.tracers.jvm.JVMStatisticsTracer;
+import com.criteo.hadoop.garmadon.agent.tracers.presto.PluginClassLoaderTracer;
+
+import java.lang.instrument.Instrumentation;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+public class PrestoModule implements GarmadonAgentModule {
+
+    @Override
+    public void setup(Instrumentation instrumentation, AsyncEventProcessor eventProcessor) {
+        ExecutorService executorService = Executors.newFixedThreadPool(5);
+        // JVM/GC metrics/events
+        executorService.submit(() -> JVMStatisticsTracer.setup(
+                (timestamp, event) -> eventProcessor.offer(timestamp, StandaloneHeader.getInstance().getHeader(), event)));
+
+        // Byte code instrumentation
+        executorService.submit(() -> PluginClassLoaderTracer.setup(instrumentation));
+        executorService.submit(() -> FileSystemTracer.setup(instrumentation,
+                (timestamp, event) -> eventProcessor.offer(timestamp, StandaloneHeader.getInstance().getHeader(), event)));
+
+        executorService.shutdown();
+        // We wait 3 sec executor to instrument classes
+        // If all classes are still not instrumented after that time we let the JVM continue startup
+        // in order to not block the container for too long on agent initialization
+        // Currently we are seeing avg duration of 160 s on MapRed and 6500 s on SPARK
+        // so we consider 3s as a reasonable duration
```

Comment:
slightly confused here. If we see an avg duration on map red and spark which is much much larger, why is 3s reasonable? Or am I missing something?

Reply:
My bad I should remove this comment from this module and the timeout
It is required for ContainerModule because we don't want to stuck container too long initialising the agent
On long running applications like nodemanagers, presto... delaying a few startup waiting complete initialisation of the agent is not an issue